File: PythonDigitalRiver/TransactionFactory/TransactionsFactory.py
__author__ = 'root'
import MySQLdb
import warnings
import sys
import TransactionsItemsTbfCSV
import TransactionsItemsDigRivCSV
import TransactionsItemsSettleCSV

class AbstractTransactionFactory:
    def make_transactions_db(self):
        raise NotImplementedError("""make_transactions_db
             must be defined in subclass""")


class TBFCSVFactory(AbstractTransactionFactory):
    def __init__(self):
        self.factory = TransactionsItemsTbfCSV

    def make_transactions_db(self):
        return TransactionsItemsTbfCSV()


class DigRivCSVFactory(AbstractTransactionFactory):
    def __init__(self):
        self.factory = TransactionsItemsDigRivCSV

    def make_transactions_db(self):
        return TransactionsItemsDigRivCSV()


class SettleSumCSVFactory(AbstractTransactionFactory):
    def __init__(self):
        self.factory = TransactionsItemsSettleCSV

    def make_transactions_db(self):
        return TransactionsItemsSettleCSV()


class FactoryMaker:

    def create_trans_factory(self, filename):
        try:
            exploded = filename.split(".")
            extension = exploded[1]
            if extension == "csv":
                if "tbf.csv" in filename:
                    return TBFCSVFactory
                elif "Settlement_Summary_report" in filename:
                    return SettleSumCSVFactory
                elif "Standard_settlement_Transaction_Report" in filename:
                    return DigRivCSVFactory
                else:
                    sys.stderr.write("[ERROR] File type invalid\n""")
        except MySQLdb.Error, e:
            sys.stderr.write("[ERROR] %d: %s\n""" % (e.args[0], e.args[1]))

File: DigitalRiver/TransactionFactory/TransactionsFactory.php
<?php
/**
 * Created by Timmy Ngo.
 * User: root
 * Date: 7/23/15
 * Time: 11:47 AM
 */
abstract class TransactionFactory {
    /** Function Name: makeTransactionsDB
     *  Description: Creates the transactions item object
     * @param none
     * @return A new transactions_items object
     */
    abstract function makeTransactionsDB();

}

class TBFCSVFactory extends TransactionFactory {
    /** Function Name: makeTransactionsDB
     *  Description: Creates the transactions item object
     * @param none
     * @return A new transactions_items_tbf_csv object
     */
    function makeTransactionsDB(){
        return new transactions_items_tbf_csv();
    }
}
class DigRivCSVFactory extends TransactionFactory {
    /** Function Name: makeTransactionsDB
     *  Description: Creates the transactions item object
     * @param none
     * @return A new transactions_items_DigRiv_csv object
     */
    function makeTransactionsDB(){
        return new transactions_items_DigRiv_csv();
    }
}
class SettleSumCSVFactory extends TransactionFactory {
    /** Function Name: makeTransactionsDB
     *  Description: Creates the transactions item object
     * @param none
     * @return A new transactions_items_DigRiv_Settle_csv object
     */
    function makeTransactionsDB(){
        return new transactions_items_DigRiv_Settle_csv();
    }
}

class FactoryMaker{
    /** Function Name: createFactory
     *  Description: Creates the appropriate factory for the filetype
     * @param $filename The file to make the factory for
     * @return A new factory object
     */
    public function createFactory($filename){
        $extension = explode('.',$filename);
        switch($extension[1]) {
            case 'csv' :
                switch(true){
                    case stristr($extension[0], 'tbf') :
                        return new TBFCSVFactory();
                    case stristr($extension[0], 'Settlement_Summary_Report') :
                        return new SettleSumCSVFactory();
                    case stristr($extension[0], 'Standard_Settlement_Transaction_Report'):
                        return new DigRivCSVFactory();
                    default:
                        error_log('[' . date("F j, Y, g:i a e O") . ']' .
                            "File type invalid for $extension\n", 3, "./errors.log");
                }
            default :
                error_log('[' . date("F j, Y, g:i a e O") . ']' .
                    "File type invalid for $extension\n", 3, "./errors.log");
        }
    }
}